### ISSUE TITLE
single-sign-out breaks my site (conflict?)

### ISSUE THREAD

**Issue by WorldsEndless** [solita/cas-single-sign-out#1]
I'm making a web-app in Luminus. However, as soon as I add the following line to my middleware file: 

```
(ns test-luminus.middleware
  (:require [test-luminus.layout :refer [*app-context* error-page]]
            [taoensso.timbre :as timbre]
            [environ.core :refer [env]]
            [selmer.middleware :refer [wrap-error-page]]
            [prone.middleware :refer [wrap-exceptions]]
            [ring.middleware.flash :refer [wrap-flash]]
            [immutant.web.middleware :refer [wrap-session]]
            [ring.middleware.reload :as reload]
            [ring.middleware.webjars :refer [wrap-webjars]]
            [ring.middleware.defaults :refer [site-defaults wrap-defaults]]
            [ring.middleware.anti-forgery :refer [wrap-anti-forgery]]
            [ring.middleware.format :refer [wrap-restful-format]]
            [clj-cas-client.core :refer [cas]]
            [cas-single-sign-out.middleware :refer [wrap-cas-single-sign-out]]
            )
  (:import [javax.servlet ServletContext]))       
```

trying to load any page in a browser dies. The stack trace is: 

```
15-Nov-02 10:16:50 user42 INFO [test-luminus.core] - server started on port: 3000
15-Nov-02 10:16:59 user42 ERROR [test-luminus.middleware] - 
                                                                    java.lang.Thread.run                    Thread.java:  745
                                      java.util.concurrent.ThreadPoolExecutor$Worker.run        ThreadPoolExecutor.java:  617
                                       java.util.concurrent.ThreadPoolExecutor.runWorker        ThreadPoolExecutor.java: 1142
                                             io.undertow.server.HttpServerExchange$1.run        HttpServerExchange.java:  778
                                        io.undertow.server.Connectors.executeRootHandler                Connectors.java:  199
                       io.undertow.server.session.SessionAttachmentHandler.handleRequest  SessionAttachmentHandler.java:   68
org.projectodd.wunderboss.web.undertow.async.websocket.UndertowWebsocket$2.handleRequest         UndertowWebsocket.java:  103
                  immutant.web.internal.undertow/create-http-handler/reify/handleRequest                   undertow.clj:  135
                                          test-luminus.middleware/wrap-internal-error/fn                 middleware.clj:   43
                                                 test-luminus.middleware/wrap-context/fn                 middleware.clj:   38
                                  ring.middleware.x-headers/wrap-content-type-options/fn                  x_headers.clj:   53
                                         ring.middleware.x-headers/wrap-frame-options/fn                  x_headers.clj:   38
                                        ring.middleware.x-headers/wrap-xss-protection/fn                  x_headers.clj:   71
                                       ring.middleware.not-modified/wrap-not-modified/fn               not_modified.clj:   52
                                 ring.middleware.default-charset/wrap-default-charset/fn            default_charset.clj:   26
                                       ring.middleware.content-type/wrap-content-type/fn               content_type.clj:   30
                                               ring.middleware.resource/wrap-resource/fn                   resource.clj:   28
                           ring.middleware.absolute-redirects/wrap-absolute-redirects/fn         absolute_redirects.clj:   36
                                                 ring.middleware.cookies/wrap-cookies/fn                    cookies.clj:  161
                                                                                     ...                                     
                                                    robert.hooke/prepare-for-hooks/fn/fn                      hooke.clj:   54
                                                                  robert.hooke/run-hooks                      hooke.clj:   46
                                                                      clojure.core/apply                       core.clj:  630
                                                                                     ...                                     
                                                           robert.hooke/compose-hooks/fn                      hooke.clj:   40
                                                                      clojure.core/apply                       core.clj:  632
                                                                                     ...                                     
clojure.lang.ArityException: Wrong number of args (3) passed to: middleware/capture-cookies-hook

15-Nov-02 10:16:59 user42 ERROR [test-luminus.middleware] - 
                                                                    java.lang.Thread.run                    Thread.java:  745
                                      java.util.concurrent.ThreadPoolExecutor$Worker.run        ThreadPoolExecutor.java:  617
                                       java.util.concurrent.ThreadPoolExecutor.runWorker        ThreadPoolExecutor.java: 1142
                                             io.undertow.server.HttpServerExchange$1.run        HttpServerExchange.java:  778
                                        io.undertow.server.Connectors.executeRootHandler                Connectors.java:  199
                       io.undertow.server.session.SessionAttachmentHandler.handleRequest  SessionAttachmentHandler.java:   68
org.projectodd.wunderboss.web.undertow.async.websocket.UndertowWebsocket$2.handleRequest         UndertowWebsocket.java:  103
                  immutant.web.internal.undertow/create-http-handler/reify/handleRequest                   undertow.clj:  135
                                          test-luminus.middleware/wrap-internal-error/fn                 middleware.clj:   43
                                                 test-luminus.middleware/wrap-context/fn                 middleware.clj:   38
                                  ring.middleware.x-headers/wrap-content-type-options/fn                  x_headers.clj:   53
                                         ring.middleware.x-headers/wrap-frame-options/fn                  x_headers.clj:   38
                                        ring.middleware.x-headers/wrap-xss-protection/fn                  x_headers.clj:   71
                                       ring.middleware.not-modified/wrap-not-modified/fn               not_modified.clj:   52
                                 ring.middleware.default-charset/wrap-default-charset/fn            default_charset.clj:   26
                                       ring.middleware.content-type/wrap-content-type/fn               content_type.clj:   30
                                               ring.middleware.resource/wrap-resource/fn                   resource.clj:   28
                           ring.middleware.absolute-redirects/wrap-absolute-redirects/fn         absolute_redirects.clj:   36
                                                 ring.middleware.cookies/wrap-cookies/fn                    cookies.clj:  161
                                                                                     ...                                     
                                                    robert.hooke/prepare-for-hooks/fn/fn                      hooke.clj:   54
                                                                  robert.hooke/run-hooks                      hooke.clj:   46
                                                                      clojure.core/apply                       core.clj:  630
                                                                                     ...                                     
                                                           robert.hooke/compose-hooks/fn                      hooke.clj:   40
                                                                      clojure.core/apply                       core.clj:  632
                                                                                     ...                                     
clojure.lang.ArityException: Wrong number of args (3) passed to: middleware/capture-cookies-hook

15-Nov-02 10:16:59 user42 ERROR [test-luminus.middleware] - 
                                                                    java.lang.Thread.run                    Thread.java:  745
                                      java.util.concurrent.ThreadPoolExecutor$Worker.run        ThreadPoolExecutor.java:  617
                                       java.util.concurrent.ThreadPoolExecutor.runWorker        ThreadPoolExecutor.java: 1142
                                             io.undertow.server.HttpServerExchange$1.run        HttpServerExchange.java:  778
                                        io.undertow.server.Connectors.executeRootHandler                Connectors.java:  199
                       io.undertow.server.session.SessionAttachmentHandler.handleRequest  SessionAttachmentHandler.java:   68
org.projectodd.wunderboss.web.undertow.async.websocket.UndertowWebsocket$2.handleRequest         UndertowWebsocket.java:  103
                  immutant.web.internal.undertow/create-http-handler/reify/handleRequest                   undertow.clj:  135
                                          test-luminus.middleware/wrap-internal-error/fn                 middleware.clj:   43
                                                 test-luminus.middleware/wrap-context/fn                 middleware.clj:   38
                                  ring.middleware.x-headers/wrap-content-type-options/fn                  x_headers.clj:   53
                                         ring.middleware.x-headers/wrap-frame-options/fn                  x_headers.clj:   38
                                        ring.middleware.x-headers/wrap-xss-protection/fn                  x_headers.clj:   71
                                       ring.middleware.not-modified/wrap-not-modified/fn               not_modified.clj:   52
                                 ring.middleware.default-charset/wrap-default-charset/fn            default_charset.clj:   26
                                       ring.middleware.content-type/wrap-content-type/fn               content_type.clj:   30
                                               ring.middleware.resource/wrap-resource/fn                   resource.clj:   28
                           ring.middleware.absolute-redirects/wrap-absolute-redirects/fn         absolute_redirects.clj:   36
                                                 ring.middleware.cookies/wrap-cookies/fn                    cookies.clj:  161
                                                                                     ...                                     
                                                    robert.hooke/prepare-for-hooks/fn/fn                      hooke.clj:   54
                                                                  robert.hooke/run-hooks                      hooke.clj:   46
                                                                      clojure.core/apply                       core.clj:  630
                                                                                     ...                                     
                                                           robert.hooke/compose-hooks/fn                      hooke.clj:   40
                                                                      clojure.core/apply                       core.clj:  632
                                                                                     ...                                     
clojure.lang.ArityException: Wrong number of args (3) passed to: middleware/capture-cookies-hook

15-Nov-02 10:16:59 user42 ERROR [test-luminus.middleware] - 
                                                                    java.lang.Thread.run                    Thread.java:  745
                                      java.util.concurrent.ThreadPoolExecutor$Worker.run        ThreadPoolExecutor.java:  617
                                       java.util.concurrent.ThreadPoolExecutor.runWorker        ThreadPoolExecutor.java: 1142
                                             io.undertow.server.HttpServerExchange$1.run        HttpServerExchange.java:  778
                                        io.undertow.server.Connectors.executeRootHandler                Connectors.java:  199
                       io.undertow.server.session.SessionAttachmentHandler.handleRequest  SessionAttachmentHandler.java:   68
org.projectodd.wunderboss.web.undertow.async.websocket.UndertowWebsocket$2.handleRequest         UndertowWebsocket.java:  103
                  immutant.web.internal.undertow/create-http-handler/reify/handleRequest                   undertow.clj:  135
                                          test-luminus.middleware/wrap-internal-error/fn                 middleware.clj:   43
                                                 test-luminus.middleware/wrap-context/fn                 middleware.clj:   38
                                  ring.middleware.x-headers/wrap-content-type-options/fn                  x_headers.clj:   53
                                         ring.middleware.x-headers/wrap-frame-options/fn                  x_headers.clj:   38
                                        ring.middleware.x-headers/wrap-xss-protection/fn                  x_headers.clj:   71
                                       ring.middleware.not-modified/wrap-not-modified/fn               not_modified.clj:   52
                                 ring.middleware.default-charset/wrap-default-charset/fn            default_charset.clj:   26
                                       ring.middleware.content-type/wrap-content-type/fn               content_type.clj:   30
                                               ring.middleware.resource/wrap-resource/fn                   resource.clj:   28
                           ring.middleware.absolute-redirects/wrap-absolute-redirects/fn         absolute_redirects.clj:   36
                                                 ring.middleware.cookies/wrap-cookies/fn                    cookies.clj:  161
                                                                                     ...                                     
                                                    robert.hooke/prepare-for-hooks/fn/fn                      hooke.clj:   54
                                                                  robert.hooke/run-hooks                      hooke.clj:   46
                                                                      clojure.core/apply                       core.clj:  630
                                                                                     ...                                     
                                                           robert.hooke/compose-hooks/fn                      hooke.clj:   40
                                                                      clojure.core/apply                       core.clj:  632

```

**Comment by noidi** [comment]
I'm sorry for the late reply. I hadn't noticed this issue due to incorrect notification settings.

Do you still have this problem?

**Comment by WorldsEndless** [comment]
Yes, including single-sign-out in my Luminus app produces 
conflicts.

**Comment by noidi** [comment]
I've released a new version, 0.1.3, that should fix this issue. Please give it a try. And thanks for the report!
